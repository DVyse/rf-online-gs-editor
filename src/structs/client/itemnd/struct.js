import Struct from '../../../classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: String, name: 'strName', len: 64 } },
]);
