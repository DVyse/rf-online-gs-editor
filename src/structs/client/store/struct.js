import Struct from '../../../classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: Number, name: 'nIndex', len: 32 } },
  { child: { type: String, name: 'strCode', len: 32, as: 'hex' } },
  { child: { type: Number, name: 'nRace', len: 8 } },
  { child: { type: String, name: 'strStoreNPCname', len: 32 } },
  { child: { type: String, name: 'strStoreNPClastName', len: 32 } },
  { child: { type: String, name: 'strModel', len: 32 } },
  { child: { type: Number, name: 'nStoreTrade', len: 8 } },
  { child: { type: Number, name: 'nUnkInt16_1', len: 16 } },
  { child: { type: Number, name: 'fStoreNPCangle', len: 32, as: 'float' } },
  { child: { type: Boolean, name: 'bSetNPCangle', len: 32 } },
  { child: { type: Number, name: 'fStoreNPCsize', len: 32, as: 'float' } },
  { child: { type: Number, name: 'nStoreLISTcount', len: 32 } },

  // listItems
  {
    child: [
      { type: Number, name: 'nItemListType', len: 32 },
      { type: String, name: 'strItemList', len: 32, as: 'hex' },
    ],
    repeat: 200,
  },

  { child: { type: Number, name: 'nNpcClass', len: 8 }, repeat: 10 },
  { child: { type: String, len: 32, as: 'hex', name: 'strUnkX32' } },
  { child: { type: Number, len: 16, name: 'nUnkInt16_2' } },
  { child: { type: Number, len: 32, name: 'nPriceSet' } },
  { child: { type: String, len: 32, name: 'strItemUpCode', as: 'hex' } },
  { child: { type: Number, len: 32, name: 'nLimitLISTcount' } },

  // limitListItems
  {
    child: [
      { type: Number, name: 'nLimItemType', len: 32 },
      { type: String, name: 'strLimItemCode', len: 32, as: 'hex' },
      { type: Number, name: 'nLimMaxCount', len: 32 },
    ],
    repeat: 16,
  },
]);
