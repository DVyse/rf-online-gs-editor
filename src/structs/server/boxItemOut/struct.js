import Struct from '~/classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: Number, name: 'nIndex', len: 32 } },
  { child: { type: String, name: 'strCode', len: 64 } },
  {
    child: [
      { type: String, len: 8, name: 'strItemCode' },
      { type: Number, len: 32, name: 'nItemCount' },
      { type: Number, len: 32, name: 'nItemProb' },
    ],
    repeat: 61,
  },
]);
