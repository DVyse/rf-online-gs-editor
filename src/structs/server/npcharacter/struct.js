import Struct from '~/classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: Number, name: 'nIndex', len: 32 } },
  { child: { type: String, name: 'strCode', len: 64 } },
  { child: { type: Number, name: 'nRace', len: 32 } },
  { child: { type: Boolean, name: 'bWarType', len: 32 } },
  { child: { type: Number, name: 'fTarDecType', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fLevel', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fExt', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttFcStd', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fMinAFSelProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fMaxAFSelProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttSklUnit', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fDefSklUnit', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fWeakPart', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fStdDefFc', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fFireTol', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fWaterTol', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fSoilTol', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fWindTol', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTechID1', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTech1UseProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTechID2', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTech2UseProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTechID3', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTech3UseProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTechMinAFProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTechMinAF', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTechMaxAFProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttTechMaxAF', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fPSecTechID', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fMSecTechID', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fMaxHP', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fHPRecDelay', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fHPRecUnit', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttSpd', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttMoTime1', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttMoTime2', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fViewExt', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAttExt', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fMRefExt', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fCopTime', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fMovSpd', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fWarMovSpd', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fScaleRate', len: 32, as: 'float' } },
  { child: { type: Boolean, name: 'bScaleChange', len: 32 } },
  { child: { type: Number, name: 'fWidth', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fWaitTime', len: 32, as: 'float' } },
  { child: { type: Number, name: 'nAsitReqRate', len: 32 } },
  { child: { type: Number, name: 'nAsitAptRate', len: 32 } },
  { child: { type: String, name: 'strChildMon', len: 64 } },
  { child: { type: Number, name: 'nChildMonNum', len: 32 } },
  { child: { type: Number, name: 'fEmoType', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fDamHPStd', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmoImpStdTime', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fGoodToOrdHPPer', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fOrdToBadHPPer', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fBadToWorseHPPer', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEspTFProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fTypeCompTerms', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fPSecTechChat', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fPAttTechChat', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo0Chat', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo0ChatProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo1Chat', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo1ChatProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo2Chat', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo2ChatProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo3Chat', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo3ChatProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo4Chat', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fEmo4ChatProb', len: 32, as: 'float' } },
  { child: { type: Number, name: 'fAsitReqSteEspChat', len: 32, as: 'float' } },
  {
    child: {
      type: Number,
      name: 'fAsitReqSteEspChatProb',
      len: 32,
      as: 'float',
    },
  },
  {
    child: { type: Number, name: 'fAsitReqSteHelpChat', len: 32, as: 'float' },
  },
  {
    child: {
      type: Number,
      name: 'fAsitReqSteHelpChatProb',
      len: 32,
      as: 'float',
    },
  },
  { child: { type: Number, name: 'fAsitReqSteCopChat', len: 32, as: 'float' } },
  {
    child: {
      type: Number,
      name: 'fAsitReqSteCopChatProb',
      len: 32,
      as: 'float',
    },
  },
]);
