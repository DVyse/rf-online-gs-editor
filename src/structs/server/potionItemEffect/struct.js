import Struct from '~/classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: Number, name: 'nIndex', len: 32 } },
  { child: { type: String, name: 'strCode', len: 64 } },
  { child: { type: Number, name: 'nClass', len: 32 } },
  { child: { type: Number, name: 'nIconIDX', len: 32 } },
  { child: { type: Number, name: 'nMastIndex', len: 32 } },
  { child: { type: String, name: 'strMastKorName', len: 64 } },
  { child: { type: String, name: 'strMastEngName', len: 64 } },
  { child: { type: String, name: 'strKorName', len: 64 } },
  { child: { type: String, name: 'strEngName', len: 64 } },
  { child: { type: Number, name: 'nLv', len: 32 } },
  { child: { type: Boolean, name: 'bActivate', len: 32 } },
  { child: { type: Boolean, name: 'bEnable', len: 32 } },
  { child: { type: String, name: 'strUsableRace', len: 64 } },
  { child: { type: String, name: 'strActableDst', len: 64 } },
  { child: { type: String, name: 'strGradeLimit', len: 64 } },
  { child: { type: Number, name: 'nNeedMastIndex', len: 32 } },
  { child: { type: String, name: 'strFixWeapon', len: 64 } },
  { child: { type: Boolean, name: 'bFixshield', len: 32 } },
  { child: { type: Number, name: 'nSpecialType', len: 32 } },
  { child: { type: Number, name: 'nNeedSpecialType', len: 32 } },
  { child: { type: Number, name: 'nNeedHP', len: 32 } },
  { child: { type: Number, name: 'nNeedFP', len: 32 } },
  { child: { type: Number, name: 'nNeedSP', len: 32 } },

  // _consume_item_list
  {
    child: [
      {
        type: String,
        name: 'itmNeedItemCode',
        len: 8,
      },
      { type: Number, name: 'nNeedItemCount', len: 32 },
    ],
    repeat: 3,
  },

  { child: { type: Number, name: 'fActDelay', len: 32, as: 'float' } },
  { child: { type: Boolean, name: 'bCumulType', len: 32 } },
  { child: { type: Number, name: 'nCumulCounter', len: 32 } },
  { child: { type: Number, name: 'nNewEffCount', len: 32 } },
  { child: { type: String, name: 'strEffectCode', len: 64 } },
  { child: { type: Number, name: 'nAttackable', len: 32 } },
  { child: { type: Number, name: 'nAttType', len: 32 }, repeat: 7 },
  { child: { type: Number, name: 'nAttConstant', len: 32 }, repeat: 7 },
  {
    child: { type: Number, name: 'fAttFormulaConstant', len: 32, as: 'float' },
  },
  { child: { type: Number, name: 'nAttNeedBt', len: 32 } },
  { child: { type: Number, name: 'nBonusDistance', len: 32 } },
  { child: { type: String, name: 'strRangeEffCode', len: 64 } },
  { child: { type: Number, name: 'nTempEffectType', len: 32 } },
  { child: { type: Number, name: 'nTempParamCode', len: 32 } },
  {
    child: { type: Number, name: 'fTempValue', len: 32, as: 'float' },
    repeat: 7,
  },
  { child: { type: Number, name: 'nContEffectType', len: 32 } },
  { child: { type: Number, name: 'nEffLimType', len: 32 } },
  { child: { type: Number, name: 'nEffLimType2', len: 32 } },
  { child: { type: Number, name: 'nContAreaType', len: 32 } },

  // _cont_param_list
  {
    child: [
      { type: Number, name: 'nContParamCode', len: 32 },
      { type: Number, name: 'nContParamIndex', len: 32 },
      {
        child: { type: Number, name: 'fContValue', len: 32, as: 'float' },
        repeat: 7,
      },
    ],
    repeat: 5,
  },

  { child: { type: Number, name: 'nContEffectSec', len: 32 }, repeat: 7 },
  { child: { type: Number, name: 'nEtc', len: 32 } },
  { child: { type: Number, name: 'f1s2speed', len: 32, as: 'float' } },
  { child: { type: Number, name: 'f1s2distance', len: 32, as: 'float' } },
  { child: { type: Number, name: 'f2s3speed', len: 32, as: 'float' } },
  { child: { type: Number, name: 'f2s3distance', len: 32, as: 'float' } },
  { child: { type: Number, name: 'nEffectClass', len: 32 } },
]);
