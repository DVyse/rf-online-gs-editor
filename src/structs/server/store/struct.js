import Struct from '../../../classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: Number, name: 'nIndex', len: 32 } },
  { child: { type: String, name: 'strCode', len: 64 } },
  { child: { type: String, name: 'strBindingDummyName', len: 64 } },
  { child: { type: String, name: 'strStoreNPCcode', len: 64 } },
  { child: { type: String, name: 'strStoreNPCname', len: 64 } },
  { child: { type: String, name: 'strStoreMAPcode', len: 64 } },
  { child: { type: Number, name: 'nStoreTrade', len: 32 } },
  { child: { type: Boolean, name: 'bSetNPCangle', len: 32 } },
  { child: { type: Number, name: 'nStoreNPCangle', len: 32 } },
  { child: { type: Number, name: 'nNpcClass', len: 32 }, repeat: 10 },
  { child: { type: Number, name: 'nStoreListCount', len: 32 } },
  { child: { type: Number, name: 'nLimitListCount', len: 32 } },
  { child: { type: Number, name: 'nLimitItemInitTime', len: 32 } },
  { child: { type: Number, name: 'nPriceSet', len: 32 } },
  { child: { type: Number, name: 'nItemUpCode', len: 32 } },
  {
    child: { type: String, name: 'strItemCode', len: 64 },
    repeat: 200,
  },
  {
    child: [
      { type: String, name: 'strLimItemCode', len: 64 },
      { type: Number, name: 'nLimMaxCount', len: 32 },
    ],
    repeat: 16,
  },
]);
