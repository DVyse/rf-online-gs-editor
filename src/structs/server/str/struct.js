import Struct from '~/classes/Struct';

export default new Struct().fromSchema1([
  { child: { type: Number, name: 'nIndex', len: 32 } },
  { child: { type: String, name: 'strCode', len: 64 } },
  { child: { type: String, name: 'strKo', len: 64 } },
  { child: { type: String, name: 'strBr', len: 64 } },
  { child: { type: String, name: 'strCh', len: 64 } },
  { child: { type: String, name: 'strEn', len: 64 } },
  { child: { type: String, name: 'strIn', len: 64 } },
  { child: { type: String, name: 'strJa', len: 64 } },
  { child: { type: String, name: 'strPh', len: 64 } },
  { child: { type: String, name: 'strRu', len: 64 } },
  { child: { type: String, name: 'strTa', len: 64 } },
  { child: { type: String, name: 'strVi', len: 64 } },
  { child: { type: String, name: 'strGlobal', len: 64 } },
]);
