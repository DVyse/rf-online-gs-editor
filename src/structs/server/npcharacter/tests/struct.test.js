import struct from '../struct';

describe('Structs', () => {
  describe('Server', () => {
    describe('NPCharacter', () => {
      describe('Struct', () => {
        it('should exist', () => {
          expect(struct.toPlainObject()).toMatchSnapshot();
        });
      });
    });
  });
});
