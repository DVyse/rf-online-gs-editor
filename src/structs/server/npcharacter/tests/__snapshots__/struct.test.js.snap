// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structs Server NPCharacter Struct should exist 1`] = `
Object {
  "fieldNames": Array [
    "nIndex",
    "strCode",
    "strName",
    "nRace",
    "bWarType",
    "fTarDecType",
    "fLevel",
    "fExt",
    "fAttFcStd",
    "fMinAFSelProb",
    "fMaxAFSelProb",
    "fAttSklUnit",
    "fDefSklUnit",
    "fWeakPart",
    "fStdDefFc",
    "fFireTol",
    "fWaterTol",
    "fSoilTol",
    "fWindTol",
    "fAttTechID1",
    "fAttTech1UseProb",
    "fAttTechID2",
    "fAttTech2UseProb",
    "fAttTechID3",
    "fAttTech3UseProb",
    "fAttTechMinAFProb",
    "fAttTechMinAF",
    "fAttTechMaxAFProb",
    "fAttTechMaxAF",
    "fPSecTechID",
    "fMSecTechID",
    "fMaxHP",
    "fHPRecDelay",
    "fHPRecUnit",
    "fAttSpd",
    "fAttMoTime1",
    "fAttMoTime2",
    "fViewExt",
    "fAttExt",
    "fMRefExt",
    "fCopTime",
    "fMovSpd",
    "fWarMovSpd",
    "fScaleRate",
    "bScaleChange",
    "fWidth",
    "fWaitTime",
    "nAsitReqRate",
    "nAsitAptRate",
    "strChildMon",
    "nChildMonNum",
    "fEmoType",
    "fDamHPStd",
    "fEmoImpStdTime",
    "fGoodToOrdHPPer",
    "fOrdToBadHPPer",
    "fBadToWorseHPPer",
    "fEspTFProb",
    "fTypeCompTerms",
    "fPSecTechChat",
    "fPAttTechChat",
    "fEmo0Chat",
    "fEmo0ChatProb",
    "fEmo1Chat",
    "fEmo1ChatProb",
    "fEmo2Chat",
    "fEmo2ChatProb",
    "fEmo3Chat",
    "fEmo3ChatProb",
    "fEmo4Chat",
    "fEmo4ChatProb",
    "fAsitReqSteEspChat",
    "fAsitReqSteEspChatProb",
    "fAsitReqSteHelpChat",
    "fAsitReqSteHelpChatProb",
    "fAsitReqSteCopChat",
    "fAsitReqSteCopChatProb",
  ],
  "fields": Array [
    Object {
      "as": undefined,
      "len": 32,
      "name": "nIndex",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strCode",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strName",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nRace",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "bWarType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Boolean",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTarDecType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fLevel",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fExt",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttFcStd",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fMinAFSelProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fMaxAFSelProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttSklUnit",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fDefSklUnit",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fWeakPart",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fStdDefFc",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fFireTol",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fWaterTol",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fSoilTol",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fWindTol",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTechID1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTech1UseProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTechID2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTech2UseProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTechID3",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTech3UseProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTechMinAFProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTechMinAF",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTechMaxAFProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttTechMaxAF",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fPSecTechID",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fMSecTechID",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fMaxHP",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fHPRecDelay",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fHPRecUnit",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttSpd",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttMoTime1",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttMoTime2",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fViewExt",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAttExt",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fMRefExt",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fCopTime",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fMovSpd",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fWarMovSpd",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fScaleRate",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "bScaleChange",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Boolean",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fWidth",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fWaitTime",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAsitReqRate",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nAsitAptRate",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": undefined,
      "len": 64,
      "name": "strChildMon",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "String",
      "weight": 64,
    },
    Object {
      "as": undefined,
      "len": 32,
      "name": "nChildMonNum",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmoType",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fDamHPStd",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmoImpStdTime",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fGoodToOrdHPPer",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fOrdToBadHPPer",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fBadToWorseHPPer",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEspTFProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fTypeCompTerms",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fPSecTechChat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fPAttTechChat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo0Chat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo0ChatProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo1Chat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo1ChatProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo2Chat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo2ChatProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo3Chat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo3ChatProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo4Chat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fEmo4ChatProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAsitReqSteEspChat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAsitReqSteEspChatProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAsitReqSteHelpChat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAsitReqSteHelpChatProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAsitReqSteCopChat",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
    Object {
      "as": "float",
      "len": 32,
      "name": "fAsitReqSteCopChatProb",
      "schemaProps": Object {},
      "type": [Function],
      "typeClassName": "Number",
      "weight": 4,
    },
  ],
  "isDynamic": false,
  "weight": 488,
}
`;
